docs/reference/module-api/module-overview.md

```diff
@@ -483,18 +483,354 @@ module.exports = {
+
+The name of the route dictates the template file that will be rendered. For example the `latest` route in the `product` module's `renderRoutes` section will return the template at `/modules/product/views/latest.html`.
+
+Information returned by the component function will be used in the associated template as `data`.
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  renderRoutes(self) {
+    return {
+      get: {
+        // GET /api/v1/product/latest
+        // The route rendered HTML for /modules/product/views/latest.html
+        async latest(req) {
+          const products = await self.find(req)
+            .sort({ createdAt: -1 })
+            .limit(req.query.max || 5)
+            .toArray();
+
+          return {
+            products
+          };
+        }
+      }
+    };
+  }
+};
+```
+
 ### `routes(self)`
+
+Add standard Express routes. The `routes` function takes takes the module as an argument and must return an object with properties for the relevant [HTTP request method](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods), including `get`, `post`, `patch`, and `delete`. Each of those properties should be set to an object of functions.
+
+Each `routes` function takes the Express arguments `req` (the [request object](https://expressjs.com/en/api.html#req)) and `res` (the [response object](https://expressjs.com/en/api.html#res)). The functions must return using the `res.redirect` or `res.send` methods.
+
+See [Naming routes](#naming-routes) for more on function names and their route URLs.
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  renderRoutes(self) {
+    return {
+      get: {
+        // GET /api/v1/product/redirect/:_id
+        async redirect(req, res) {
+          const product = await self.find(req).toObject();
+
+          return res.redirect(product._url);
+        }
+      }
+    };
+  }
+};
+```
+
 ### `handlers(self)`
+
+The `handlers` function takes the module as an argument and must return an object. The object keys should be names of existing server-side events. The value of those event keys should be an object of functions to execute when those events fire.
+<!-- TODO: Link to the reference to or guide on server-side events when available. -->
+
+Events belonging to the same module where the handlers are defined, or from its base class, can be referenced simply by name, e.g., `beforeInsert` for any piece type. You may also add handlers in one module that respond to events in other modules. Those event names should be prefixed with the name of the module where the event fires followed by a colon, e.g., `@apostrophecms/page:beforeSend`.
```

Comment:
Event handlers may be async functions, and they will be awaited before execution continues.

Reply:
I've been internally debating how much to explicitly say that things can be async. It's probably a good idea to do more.